### PR TITLE
[Fix] rubocop builds

### PR DESCRIPTION
**Pull request by grahamjenson**
removing the `env -i` so we can build native gems.

### CODE REVIEW

File: ruby/private/toolchains/repository_context.bzl [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #
 
 def _eval_ruby(ruby, script, options = None):
-    arguments = ["env", "-i", ruby.interpreter_realpath]
+    arguments = [ruby.interpreter_realpath]
```
Comment (yugui) [inline ruby/private/toolchains/repository_context.bzl]: Did you really need to remove `env -i` from this place?

File: ruby/private/tools/repository_context.bzl [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #
 
 def _eval_ruby(ruby, script, options = None):
-    arguments = ["env", "-i", ruby.interpreter_realpath]
+    arguments = [ruby.interpreter_realpath]
```
Comment (yugui) [inline ruby/private/tools/repository_context.bzl]: ditto.

File: ruby/private/toolchains/repository_context.bzl (hunk 1 above)
Comment (kigster) [reply ruby/private/toolchains/repository_context.bzl]: I think this was an experiment to see if building gems with native extensions on Linux was possible. Our theory was that env -i was responsible for crashes during compilation of the native gems on CI.

I didn’t realize this was merged so quickly, but I have a different theory as to why the gems aren’t building natively on CI, but they do build locally on OSX. If my theory proves correct we may want to revert this PR.

My current theory is that the native extensions compilation together with the aggressive java flags resulted in out of memory error. I had to fight it yesterday when implementing circleci. Let me find a reference....

Comment (kigster) [reply ruby/private/toolchains/repository_context.bzl]: This is the error we've been seeing on Linux / CI only:

https://stackoverflow.com/questions/11912878/gcc-error-gcc-error-trying-to-exec-cc1-execvp-no-such-file-or-directory

There are any answers on that page but none that mention out of memory error. And then I found a link from Alex Eagle about this issue and saying that have argrs should be set lower and number of jobs lower. Let me confirm that this indeed resolves the issue without removing env -i

**Pull request merged**